### ISSUE TITLE
bud: `COPY --from=` uses wrong layer

### ISSUE THREAD

**Issue by jamescassell** [containers/buildah#2415]
**Description**
Buildah chooses the wrong layer to copy from if a layer has already been used in another "FROM" command that made any modifications in its own copy of that layer.  It can be worked around by adding an extra step in the Dockerfile, but it gets unweildy.

**Steps to reproduce the issue:**
1. `podman build -f test.Dockerfile`
2. `podman diff <built-image>`
3. observe "derived" in output, not "layer"

test.Dockerfile
```
FROM registry.fedoraproject.org/fedora:32 as layer
RUN touch /root/layer

FROM layer as derived
RUN touch /root/derived ; rm /root/layer

FROM registry.fedoraproject.org/fedora:32 as output
COPY --from=layer /root /root
```

Interestingly, you can work around it by adding an extra FROM line:
workaround.Dockerfile:
```
FROM registry.fedoraproject.org/fedora:32 as layer
RUN touch /root/layer

FROM layer as derived
RUN touch /root/derived ; rm /root/layer

FROM layer
FROM registry.fedoraproject.org/fedora:32 as output
COPY --from=layer /root /root
```

**Describe the results you received:**
final image has copied from "derived", not from "layer"

```
C /root
A /root/derived
```

**Describe the results you expected:**
final image has copied from "layer" as specified in dockerfile

```
C /root
A /root/layer
```

**Output of `rpm -q buildah` or `apt list buildah`:**

```
buildah-1.14.9-1.fc32.x86_64
```

**Output of `buildah version`:**

```
Version:         1.14.9
Go Version:      go1.14.2
Image Spec:      1.0.1-dev
Runtime Spec:    1.0.1-dev
CNI Spec:        0.4.0
libcni Version:  
image Version:   5.4.3
Git Commit:      
Built:           Wed Dec 31 19:00:00 1969
OS/Arch:         linux/amd64
```

**Output of `podman version` if reporting a `podman build` issue:**

```
(paste your output here)Version:            1.9.3
RemoteAPI Version:  1
Go Version:         go1.14.2
OS/Arch:            linux/amd64

```

**Output of `cat /etc/*release`:**

```
(paste your output here)NAME=Fedora
VERSION="32 (Workstation Edition)"
ID=fedora
VERSION_ID=32
VERSION_CODENAME=""
PLATFORM_ID="platform:f32"
PRETTY_NAME="Fedora 32 (Workstation Edition)"
ANSI_COLOR="0;34"
LOGO=fedora-logo-icon
CPE_NAME="cpe:/o:fedoraproject:fedora:32"
HOME_URL="https://fedoraproject.org/"
DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora/f32/system-administrators-guide/"
SUPPORT_URL="https://fedoraproject.org/wiki/Communicating_and_getting_help"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=32
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=32
PRIVACY_POLICY_URL="https://fedoraproject.org/wiki/Legal:PrivacyPolicy"
VARIANT="Workstation Edition"
VARIANT_ID=workstation
Fedora release 32 (Thirty Two)
Fedora release 32 (Thirty Two)

```

**Output of `uname -a`:**

```
(paste your output here)Linux localhost.localdomain 5.6.6-300.fc32.x86_64 #1 SMP Tue Apr 21 13:44:19 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

```

**Output of `cat /etc/containers/storage.conf`:**

```
(paste your output here)grep -v -e '^#' -e'^$'  /etc/containers/storage.conf 
[storage]
driver = "overlay"
runroot = "/var/run/containers/storage"
graphroot = "/var/lib/containers/storage"
[storage.options]
additionalimagestores = [
]
[storage.options.overlay]
mountopt = "nodev,metacopy=on"
[storage.options.thinpool]

```

**Comment by nalind** [comment]
Doing a spot-check with the master and release-v1.14 tips, it looks like this is fixed by #2401/#2321.

**Comment by jamescassell** [comment]
thanks! seems fixed in `podman-2.0.0-0.3.rc7.fc32.x86_64`
